common-lib/src/test/scala/com/gu/mediaservice/lib/aws/ContentDispositionTest.scala

```diff
@@ -0,0 +1,37 @@
+package com.gu.mediaservice.lib.aws
+
+import com.gu.mediaservice.lib.elasticsearch.MappingTest
+import com.gu.mediaservice.model.{Image, Thumbnail}
+import org.scalatest.funsuite.AnyFunSuiteLike
+import org.scalatest.matchers.should.Matchers.convertToAnyShouldWrapper
+
+class ContentDispositionTest extends AnyFunSuiteLike with ContentDisposition {
+
+  test("build content disposition header based on filename") {
+    val image = withFilename(MappingTest.testImage, "imagefilename.jpg")
+
+    val header = getContentDisposition(image, Thumbnail, shortenDownloadFilename = false)
+
+    header shouldBe """attachment; filename="abcdef1234567890"; filename*=UTF-8''imagefilename%20%28abcdef1234567890%29.jpg"""
```

Comment:
nitpicking, but I think it'd make sense if the latin1 fallback included the extension

```suggestion
    header shouldBe """attachment; filename="abcdef1234567890.jpg"; filename*=UTF-8''imagefilename%20%28abcdef1234567890%29.jpg"""
```

Reply:
Tend to agree. The current implementation doesn't.
Deja vu about the extension not been available near this code but I'll look.

Reply:
Added a new commit to implement this.